util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -140,10 +140,13 @@ public static ApacheHttpTransport createApacheHttpTransport(
         proxyUri != null || proxyCredentials == null,
         "if proxyUri is null than proxyCredentials should be null too");
 
-    HttpClientBuilder httpClientBuilder =
-        ApacheHttpTransport.newDefaultHttpClientBuilder()
-            .setProxy(
-                proxyUri == null ? null : new HttpHost(proxyUri.getHost(), proxyUri.getPort()));
+    HttpClientBuilder httpClientBuilder = ApacheHttpTransport.newDefaultHttpClientBuilder();
+
+    if (proxyUri != null) {
+      httpClientBuilder
+          .setRoutePlanner(null)
```

Comment:
we might want to add a comment here to indicating why we need to set RoutePlanner to `null`

Reply:
Done.